src/vanilla.ts

```diff
@@ -245,9 +245,7 @@ const buildProxyFunction = (
           value = getUntracked(value) || value
         }
         let nextValue = value
-        if (Object.getOwnPropertyDescriptor(target, prop)?.set) {
```

Comment:
Nice! I had came across this code and couldn't really piece together why it was necessary; i.e. why we wanted to avoid the promise/child proxy logic just b/c this was a setter.

I am a little morbidly curious why it was necessary in the 1st place, but that's not super important, and great to see it go away.

Reply:
I wanted to avoid handling things if setter exists, because we don't know what the setter does. It can be something really unexpected. But, anyway, this would simplify it. Not really sure what happens...